rose/client/main.py

```diff
@@ -72,8 +72,18 @@ def main():
         print 'usage: rose-client drive-module'
         sys.exit(2)
 
-    module_name = splitext(sys.argv[1])[0]
-    driver_mod = import_module(module_name)
+    file_name = sys.argv[1]
+    module_path, file_suffix = os.path.splitext(file_name)
+    module_name = os.path.basename(module_path)
+
+    if file_suffix not in (".py", ".pyc"):
+        print "driver file must be either .py or .pyc"
+        sys.exit(3)
+
+    if file_suffix == ".py":
+        driver_mod = imp.load_source(module_name, file_name)
+    else:
+        driver_mod = imp.load_compiled(module_name, file_name)
```

Comment:
This mostly works (does not support .pyo files), but why bother with the different file formats, when we can use the higher level apis like imp.find_module() and imp.load_module()? In this case imp module does the right thing for us.

See https://docs.python.org/2/library/imp.html#examples

Also, the code for loading the driver module should be extracted to a separate function to make it easy to test.

Reply:
👍 Thanks for the productive feedback.
Now using imp.find_module and imp.load_module.
I extracted to load_driver_module, I'm not sure though if the expected behavior of this function is to return None or raise an exception.